os/android/drmhwctwo.cpp

```diff
@@ -670,8 +670,19 @@ HWC2::Error DrmHwcTwo::HwcLayer::SetLayerSourceCrop(hwc_frect_t crop) {
 
 HWC2::Error DrmHwcTwo::HwcLayer::SetLayerSurfaceDamage(hwc_region_t damage) {
   supported(__func__);
-  // TODO: We don't use surface damage, marking as unsupported
-  unsupported(__func__, damage);
+
+  std::vector<hwcomposer::HwcRect<int>> hwc_rects;
+
+  for (size_t rect = 0; rect < damage.numRects; ++rect) {
+    hwc_rects.push_back({damage.rects[rect].left, damage.rects[rect].top,
```

Comment:
HwcRegion and hwc_region_t is not of one to one mapping. The hwc_rect_t and the HwcRect are not matching.